example(vercel):Add vercel connector example (#252)
fivetran/fivetran_connector_sdk: examples/source_examples/vercel/README.md

## Team access
By default, the connector accesses resources from your personal Vercel account. To access deployments owned by a team:

1. Find your Team ID by navigating to your team settings in the Vercel dashboard or use the Vercel CLI.
2. Add the `team_id` parameter to the connector's `configuration.json` file.
3. Ensure your API token has the necessary permissions to access the team's resources.

When `team_id` is provided, the connector appends `?teamId=[teamID]` to the API endpoint URL to access team resources instead of personal account resources.

## Pagination
The connector implements Vercel's timestamp-based pagination system as described in the API documentation. It processes data in configurable batches and uses the `next` timestamp from the pagination response to fetch subsequent pages.

## Data handling
The connector processes data from the `/v6/deployments` endpoint which contains deployment records with status, timing, and build information. All nested JSON structures are flattened to create optimal table schemas for your destination. 

## Error handling
The connector implements comprehensive error handling strategies:

> - HTTP timeout handling (30 seconds)

Reviewer: ```suggestion
- A 30-second HTTP timeout handling
```

## Requirements
- [Supported Python versions](https://github.com/fivetran/fivetran_connector_sdk/blob/main/README.md#requirements)
- Operating system:
    - Windows: 10 or later (64-bit only)
    - macOS: 13 (Ventura) or later (Apple Silicon [arm64] or Intel [x86_64])
    - Linux: Distributions such as Ubuntu 20.04 or later, Debian 10 or later, or Amazon Linux 2 or later (arm64 or x86_64)

## Getting started
Refer to the [Connector SDK Setup Guide](https://fivetran.com/docs/connectors/connector-sdk/setup-guide) to get started.

## Features
- Synchronizes deployments from Vercel API `/v6/deployments` endpoint
- Supports both personal account and team resource access
- Implements pagination handling for large datasets with configurable limit 
- Incremental synchronization using timestamp-based state management 
- Comprehensive error handling with retry logic for API requests

## Configuration file

> The configuration keys required for your connector are:

Reviewer: ```suggestion
The configuration keys required for your connector are as follows:
```

```json
{
  "api_token": "YOUR_VERCEL_API_TOKEN",
  "team_id": "<YOUR_TEAM_ID_OPTIONAL>"
}
```

> ### Configuration Parameters

Reviewer: ```suggestion
### Configuration parameters
```

Author: I feel the above may not be required because we already have 
`The configuration keys required for your connector are as follows:` and then we explain about each of the Configuration parameters. Please let me know what you think?

> - **`api_token`** (required): Your Vercel Access Token for API authentication

Reviewer: ```suggestion
- `api_token` (required): Your Vercel access token for API authentication
```

- **`team_id`** (optional): Team identifier to access team resources instead of personal account. If not provided, the connector will access your personal account resources.

Note: Ensure that the `configuration.json` file is not checked into version control to protect sensitive information.

## Requirements file
The connector uses the `requests` library for HTTP communication, which is pre-installed in the Fivetran environment.

Note: The `fivetran_connector_sdk:latest` and `requests:latest` packages are pre-installed in the Fivetran environment. To avoid dependency conflicts, do not declare them in your `requirements.txt`.

## Authentication
The connector uses Vercel Access Tokens for authentication. To obtain your API token:

> 1. Navigate to your [Vercel account settings](https://vercel.com/account/tokens).

Reviewer: ```suggestion
1. Go to your [Vercel account settings](https://vercel.com/account/tokens).
```

> 2. Click "Create Token".
> 3. Enter a descriptive name for the token.
> 4. Choose the appropriate scope.
> 5. Select an expiration date.
> 6. Copy the generated token and use it as the `api_token` in your configuration.

Reviewer: ```suggestion
2. Click **Create Token**.
3. Enter a descriptive name for the token.
4. Choose the appropriate scope.
5. Select an expiration date.
6. Copy the generated token. You will use it as the `api_token` in your connector's configuration.
```
